<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MindVision Camera Viewer (Python - PySide6)</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="main_layout">
    <item>
     <layout class="QVBoxLayout" name="side_panel">
      <property name="sizeConstraint">
       <enum>QLayout::SetFixedSize</enum>
      </property>
      <item>
       <widget class="QGroupBox" name="controls_group">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Camera Settings</string>
        </property>
        <layout class="QFormLayout" name="controls_layout">
         <item row="0" column="0">
          <widget class="QCheckBox" name="chk_auto_exposure">
           <property name="text">
            <string>Auto Exposure</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QCheckBox" name="chk_roi">
           <property name="text">
            <string>High FPS ROI (640x480)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_ae_target">
           <property name="text">
            <string>Auto Exp Target:</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="spin_ae_target">
           <property name="minimum">
            <number>0</number>
           </property>
           <property name="maximum">
            <number>255</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_exposure_time">
           <property name="text">
            <string>Exposure Time:</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QDoubleSpinBox" name="spin_exposure_time">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="decimals">
            <number>4</number>
           </property>
           <property name="minimum">
            <double>0.100000000000000</double>
           </property>
           <property name="maximum">
            <double>1000.000000000000000</double>
           </property>
           <property name="suffix">
            <string> ms</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_exposure_slider">
           <property name="text">
            <string>Exposure Slider:</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSlider" name="slider_exposure">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="maximum">
            <number>10000</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QLabel" name="label_gain">
           <property name="text">
            <string>Analog Gain:</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QSpinBox" name="spin_gain">
           <property name="maximum">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="label_gain_slider">
           <property name="text">
            <string>Gain Slider:</string>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSlider" name="slider_gain">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="trigger_group">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="title">
         <string>Trigger Settings</string>
        </property>
        <layout class="QVBoxLayout" name="trigger_layout">
         <item>
          <widget class="QRadioButton" name="rb_continuous">
           <property name="text">
            <string>Continuous</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rb_software">
           <property name="text">
            <string>Software Trigger</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="rb_hardware">
           <property name="text">
            <string>Hardware Trigger</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="btn_soft_trigger">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <property name="text">
            <string>Trigger</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="start_btn">
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Start Camera</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stop_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Stop Camera</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="record_btn">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="maximumSize">
         <size>
          <width>300</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="text">
         <string>Start Recording</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="verticalSpacer">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>20</width>
          <height>40</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QLabel" name="video_label">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true">QLabel { background-color : black; color : white; }</string>
      </property>
      <property name="text">
       <string>Camera Stream</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>